varinfo/umm_var.py

```diff
@@ -346,53 +346,74 @@ def get_json_serializable_value(input_value: Any) -> Any:
 def publish_umm_var(collection_id: str,
                     umm_var_dict: Dict,
                     auth_header: str,
-                    cmr_env: CMR_UAT = CMR_UAT) -> Union[str, Dict]:
+                    cmr_env: CmrEnvType = CMR_UAT) -> str:
     """" Publish a single UMM-Var entry to CMR given:
         * collection_id: a collection's concept_id
         * umm_var_dict: a dictionary of a single UMM-Var entry for a collection
-        * auth_header: Earthdata Login (EDL) bearer_token or LaunchPad token
-            with their respective headers
+        * auth_header: Authorization HTTP header, containing a LaunchPad
+          token: 'Authorization: <token>'
         * cmr_env: CMR environments (OPS, UAT, and SIT) default is CMR_UAT
-    For a successful requests all of these fields must be entered
+
+        For a successful requests all of these fields must be entered
 
     """
     # Required UMM-Var headers for ingesting variable entries
     headers_umm_var = {
         'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
         + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
         'Authorization': auth_header,
-        'Accept': 'application/json'}
+        'Accept': 'application/json'
+    }
+    variable_native_id = generate_variable_native_id(collection_id,
+                                                     umm_var_dict)
 
     url_endpoint = (cmr_env.replace('search', 'ingest') + 'collections/'
-                    f'{collection_id}/variables/{umm_var_dict["Name"]}')
+                    f'{collection_id}/variables/{variable_native_id}')
 
     response = requests.put(url_endpoint,
                             json=umm_var_dict,
                             headers=headers_umm_var,
                             timeout=10)
 
-    # A successful request returns the variable concept-id
-    # e.g.'V1259791517-EEDTEST'
     if response.ok:
+        # A successful request returns the variable concept-id
+        # e.g., 'V1259791517-EEDTEST'
         return response.json()['concept-id']
-    # A failed request returns the response dict with the error message
-    # e.g. {'errors': ['#: required key [LongName] not found']}
-    return response.json()
+    else:
+        # A failed request returns the response dict with the error message
+        # e.g., {'errors': ['#: required key [LongName] not found']}
+        # This will be converted into a single string that can be returned to
+        # the end-user.
+        return '\n  '.join(response.json()['errors'])
```

Comment:
Aside from my OCD moving the comments around - I updated this to only return the error messages themselves (concatenated together into a single string).